queries/water-z9.pgsql

```diff
@@ -35,6 +35,24 @@ FROM
         AND way_area::bigint > 102400 -- 4px
         AND mz_way12 && !bbox!
 
+    --
+    -- Water line geometries
+    --
+    UNION
+
+    SELECT name,
+           NULL AS area,
+           waterway AS kind,
+           'openstreetmap.org' AS source,
+           way AS __geometry__,
+           mz_id AS __id__
+
+    FROM planet_osm_line
+
+    WHERE
+        waterway IN ('canal', 'river')
```

Comment:
I don't think we want to filter geometries that don't have a name, as those are still valid to render.

The idea of filtering out very short lines is interesting. For area filters, we are often only filtering very small geometries, so small that they are smaller than 1-2 pixels (an exception is for buildings, we filter a bit more aggressively at zoom 13 and lower since it can pull in millions of geometries in places like NYC). So theoretically if we filtered on lines smaller than 1-pixel at the current zoom, there would not be a hole or other visible effect. I am unsure how common this situation is though, ideally we'd have some data. For now we could leave them in and see if we get cases of tiles with many many small line segments?

A better, more involved approach might be pre-join/simplify the line segments into larger ones at the lower zooms. I believe @kevinkreiser has some experience with this (and may have some code for that handy). Likely outside the scope of this PR, but something to think about for the future.